src/auth/auth-api-request.ts

```diff
@@ -425,12 +472,37 @@ export const FIREBASE_AUTH_SIGN_UP_NEW_USER = new ApiSettings('signupNewUser', '
     }
   });
 
+const FIREBASE_AUTH_GET_OOB_CODE = new ApiSettings('/accounts:sendOobCode', 'POST')
+  // Set request validator.
+  .setRequestValidator((request: any) => {
+    if (!validator.isEmail(request.email)) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_EMAIL,
+      );
+    }
+    if (EMAIL_ACTION_REQUEST_TYPES.indexOf(request.requestType) === -1) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_ARGUMENT,
+        `"${request.requestType}" is not a supported email action request type.`,
+      );
+    }
+  })
+  // Set response validator.
+  .setResponseValidator((response: any) => {
+    // If the oobLink is not returned, then the request failed.
+    if (!response.oobLink) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INTERNAL_ERROR,
+        'INTERNAL ASSERT FAILED: Unable to create the email action link');
+    }
+  });
+
 /**
- * Class that provides mechanism to send requests to the Firebase Auth backend endpoints.
+ * Class that provides the mechanism to send requests to the Firebase Auth backend endpoints.
  */
 export class FirebaseAuthRequestHandler {
-  private baseUrl: string = `https://${FIREBASE_AUTH_HOST}${FIREBASE_AUTH_PATH}`;
-  private httpClient: AuthorizedHttpClient;
+  protected httpClient: AuthorizedHttpClient;
```

Comment:
Why protected?

Reply:
This will be transformed into a base class down the road. This may need to be accessed by subclasses. Removed for now.